docs/gettinginvolved/functional-testing.rst

```diff
@@ -134,7 +134,49 @@ To run the functional tests, run the following command:
 OpenStack
 =========
 
-The configuration stanza for an private OpenStack deployment is the same as Rackspace above, but ``auth_plugin`` should be included, which refers to an authentication plugin provided by ``python-keystoneclient``.
+The configuration stanza for a private OpenStack deployment is similar to Rackspace (above), with a few notable differences:
+
+* ``auth_plugin`` should be included, which refers to an authentication plugin provided by ``python-keystoneclient``, and
+* ``provider: "openstack"`` should be included, if the top level key is not ``openstack``.
 
 If required, you may need to add additional fields.
 For more information, see :ref:`openstack-dataset-backend`.
+
+Devstack
+--------
+
+It is assumed that you have a working ``Devstack`` environment.
+Refer to document "Setting up a devstack instance" on Google Drive.
+
+To run the Cinder functional tests on ``devstack``:
+
+* boot a supported guest operating system in ``devstack``,
+* log into the guest and clone your branch of the Flocker source code and
+* install the Flocker dependencies in a ``virtualenv``.
+* Create ``$HOME/acceptance.yml`` containing:
+
+.. code:: yaml
+
+   # It is important to use ``devstack-openstack`` as the top-level name
```

Comment:
:question: This looks less than clean. Trying to understand this a bit. Let me know if I'm write with my following guess:

Is this just using `devstack-openstack` instead of `cinder`? or `devstack-openstack` instead of `rackspace`?

If it's the former, this is indicative of some code badness somewhere. Specifically, cinder doesn't impose size limits on volumes, but rackspace does. Somewhere in the flocker codebase someone assumed that `cinder` meant `rackspace` and set the minimum volume size to something too large for cinder in general. Now we are just using `devstack-openstack` to work around that mistake that someone made. If I'm correct on this scenario potentially see if we could fix this in the code. Although... thinking about it now, we might not be able to given external people depend on this.

If it's the latter, we might not need this warning - devstack-openstack is just different than rackspace.